pytorch_lightning/trainer/data_loading.py

```diff
@@ -262,6 +262,10 @@ def _get_dataloader_init_kwargs(
                     "This shouldn't happen, please open an issue on Lightning Github repository."
                 )
 
+        if _inter_batch_parallelism():
+            # can only do asynchronous transfer if we use pin_memory
+            dl_kwargs["pin_memory"] = True
```

Comment:
Is this documented somewhere?

I think we should fail in case it isn't set to true instead of silently changing stuff...

Reply:
Added a warning !